src/algebra/module/dedekind_domain.lean

```diff
@@ -0,0 +1,75 @@
+/-
+Copyright (c) 2022 Pierre-Alexandre Bazin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Pierre-Alexandre Bazin
+-/
+import algebra.module.torsion
+import ring_theory.dedekind_domain.ideal
+
+/-!
+# Modules over a Dedekind domain
+
+-/
+
+universes u v
+open_locale big_operators
+
+variables {R : Type u} [comm_ring R] [is_domain R] {M : Type v} [add_comm_group M] [module R M]
+
+open_locale direct_sum
+open submodule
+
+lemma is_torsion_by_ideal_of_finite_of_is_torsion [module.finite R M] (hM : module.is_torsion R M) :
```

Comment:
Since there are no particular assumptions on `R`, this can go to a more basic file (no need to open another PR).

Reply:
indeed, this was already duplicated in `algebra.module.torsion` after merging master